frontend/src/features/Train/components/ImageUploadLayout.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  DATA_SOURCE,
+  FileUploadData,
+  ImageUploadData,
+} from "@/features/Train/types/trainTypes";
+import { Button, Radio, Stack, Typography } from "@mui/material";
+import { Controller, UseFormReturn } from "react-hook-form";
+import dynamic from "next/dynamic";
+import { formatDate } from "@/common/utils/dateFormat";
+import {
+  useGetDatasetFilesDataQuery,
+  useUploadDatasetFileMutation,
+} from "@/features/Train/redux/trainspaceApi";
+import { DataGrid } from "@mui/x-data-grid";
+import prettyBytes from "pretty-bytes";
+import React from "react";
+
+const FilerobotImageEditor = dynamic(
+  () => import("react-filerobot-image-editor"),
+  {
+    ssr: false,
+  }
+);
+
+const TABS = dynamic(
+  () => import("react-filerobot-image-editor").then((lib) => lib.TABS),
+  {
+    ssr: false,
+  }
+);
+
+const TOOLS = dynamic(
+  () => import("react-filerobot-image-editor").then((lib) => lib.TOOLS),
+  {
+    ssr: false,
+  }
+);
+
+const dataURLtoFile = (dataurl: string, filename: string) => {
+  const arr = dataurl.split(",");
+  if (arr.length === 0) {
+    return new File([""], filename);
+  }
+  const matched = arr[0].match(/:(.*?);/);
+  const mime = matched ? matched[1] : undefined;
+  const bstr = atob(arr[1]);
+  let n = bstr.length;
+  const u8arr = new Uint8Array(n);
+  while (n) {
+    u8arr[n - 1] = bstr.charCodeAt(n - 1);
+    n -= 1; // to make eslint happy
+  }
+  return new File([u8arr], filename, { type: mime });
+};
+
+export const UploadImagePanel = ({
+  dataSource,
+  methods,
+}: {
+  dataSource: DATA_SOURCE;
+  methods: UseFormReturn<ImageUploadData, unknown>;
+}) => {
+  const [uploadFile] = useUploadDatasetFileMutation();
+  const { data, refetch } = useGetDatasetFilesDataQuery({ dataSource });
+  return (
+    <>
+      {methods.formState.errors.name && (
+        <Typography>Please select a file</Typography>
+      )}
+      <Stack direction={"row"} spacing={2}>
+        <FilerobotImageEditor
+          source="https://scaleflex.airstore.io/demo/stephen-walker-unsplash.jpg"
+          onSave={(editedImageObject: unknown) => {
```

Comment:
No need to specify the type for the editedImageObject argument btw, typescript automatically infers it for you since the onSave property already specifies it